evalml/tests/component_tests/decomposer_tests/test_decomposer.py

```diff
@@ -481,3 +500,125 @@ def test_decomposer_bad_target_index(
         match="doesn't support target data with index of type",
     ):
         dec._choose_proper_index(y)
+
+
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+@pytest.mark.parametrize(
+    "transformer_fit_on_data",
+    [
+        "in-sample",
+        "in-sample-less-than-sample",
+        "wholly-out-of-sample",
+        "wholly-out-of-sample-no-gap",
+        "partially-out-of-sample",
+        "out-of-sample-in-past",
+        "partially-out-of-sample-in-past",
+    ],
+)
+def test_decomposer_fit_transform_out_of_sample(
+    decomposer_child_class,
+    generate_seasonal_data,
+    transformer_fit_on_data,
+):
+    # Generate 10 periods (the default) of synthetic seasonal data
+    seasonal_period = 7
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period=seasonal_period,
+        freq_str="D",
+        set_time_index=True,
+        seasonal_scale=0.05,  # Increasing this value causes the decomposer to miscalculate trend
```

Comment:
I added this to try and get the decomposer to perform a little better and do a better job capturing the trend.  This just reduces the amplitude of the sinusoidal oscillations added to the synthetic signal.

Reply:
Why does increasing seasonal_scale cause the decomposer to miscalculate the trend?

Reply:
That has yet to be determined.  What we've noticed in some datasets is that there is a tendency for the periodic signal to "leak" into the trend.  Loosely, I've noticed that a larger relative seasonal signal vs. trend exacerbates this to a degree.